src/fe-common/core/chat-completion.c

```diff
@@ -639,6 +639,29 @@ static void complete_window_nicks(GList **list, WINDOW_REC *window,
         }
 }
 
+/* iterates over all nicks in the list and escapes them if needed */
+static void escape_nicks(GList **list)
+{
+	GList *l;
+	char *escaped;
+
+	/* verify that the list isn't NULL */
+	g_return_if_fail(list != NULL);
+
+	/* iterate over the nick list */
+	for (l = *list; l != NULL; l = l->next) {
+		/* only escape if we find either \\r or \\n in the nick */
+		if (l->data != NULL &&
+		    (strstr(l->data, "\\r") != NULL ||
+		     strstr(l->data, "\\n") != NULL)) {
```

Comment:
Ah, yeah I was thrown by the initial issue, so I didn't think about that, rookie mistake :-).

I have removed the escape_pattern func now and the if statement you pointed out, so now it uses escape_string on all nicks.

I guess the only question now is the one below here, if you would like me to change where the escaping occurs or not. Let me know if you want me to fix this as well.